modules/hydrodyn/src/SS_Excitation.f90

```diff
@@ -77,15 +77,45 @@ subroutine TransformStateSpaceMatrices( NBody, RotZ, C )
    end do
 
 end subroutine TransformStateSpaceMatrices
+
+function GetWaveElevation ( time, u_in, t_in, p, m, ErrStat, ErrMsg )
+    real(DbKi),                       intent(in)     :: time
+    TYPE(SS_Exc_InputType),           INTENT(IN)     :: u_in(:) ! Input at t1 > t2 > t3
+    real(DbKi),                       intent(in)     :: t_in(:)
+    TYPE(SS_Exc_ParameterType),       INTENT(in)     :: p           !< Parameters      
+    TYPE(SS_Exc_MiscVarType),         INTENT(inout)  :: m           !< Initial misc/optimization variables            
+    INTEGER(IntKi),                   INTENT(  OUT)  :: ErrStat     !< Error status of the operation
+    CHARACTER(*),                     INTENT(  OUT)  :: ErrMsg      !< Error message if ErrStat /= ErrID_None
+   real(ReKi)     :: positionXY(2)
+    
+    real(SiKi)  :: GetWaveElevation(p%NBody)
+    TYPE(SS_Exc_InputType)        :: u_out  ! extra_interp result
```

Comment:
Hi,

The 2-dimensional allocatable variable `PtfmPos` of `u_out` may need to be allocated here.

Best regards,
Chen

Reply:
See fix in https://github.com/bjonkman/openfast/commit/c1b18ec739a6fb3148dfd9f41f52983889996c04 (which also fixes a few other issues with this subroutine)